### DISCUSSION TITLE
how to disable `minify` in `nuxt generate`

### DISCUSSION THREAD

**Discussion by lqzhgood** [nuxt/nuxt#23329] [Questions]
### Describe the feature

I find options to  disable `minify` in `nuxt2`
but not find in `nuxt3`.   https://nuxt.com/docs/guide/directory-structure/nuxt.config
how to  disable `minify`  in `nuxt3` ?

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by imcm7** [comment]
https://github.com/nuxt/nuxt/issues/23299#issuecomment-1727210144

```
nitro: {
    minify: false, 
},
```

**Comment by MrEraxd** [comment]
Hi @imcm7, nitro is used to create server so disabling minification here will not help.

@lqzhgood `nuxt generate` uses webpack under the hood to build. So you have to look for webpack config in nuxt. I've checked documentation and it is not mentioned here directly but you can disable it with:

```
webpack: {
    optimization: {
        minimize: false
    }
}
```

Links:
[Nuxt docs](https://nuxt.com/docs/api/configuration/nuxt-config#optimization-1)
[Disable webpack minimizer](https://davidwalsh.name/how-to-not-minify-source-with-webpack)

**Comment by lqzhgood** [comment]
@MrEraxd  3Q
but I try   `webpack:{optimization:{minimize: false}}`  options  in `nuxt.config.ts` not effect.

I also try `vite` options not effect too~
https://vitejs.dev/config/build-options.html#build-minify
```
 vite: {
        build: {
            minify: false,
        },
  }
```

 
The spaces in the `HTML` have still been removed.

**Comment by MrEraxd** [comment]
Check [this documentation](https://stackoverflow.com/questions/69055857/preserve-whitespace-between-spans-in-vue) and [this stackoverflow issue](https://stackoverflow.com/questions/69055857/preserve-whitespace-between-spans-in-vue) for preserving white space in components. Maybe this will help?
